pkg/server/http.go

```diff
@@ -252,6 +260,27 @@ func AdminAuth(expectedToken string) func(http.Handler) http.Handler {
 	}
 }
 
+// ARCAuth verifies that the ARC key is set in configuration before allowing requests.
+func ARCAuth(arc string) func(http.Handler) http.Handler {
+	type FailureResponse struct {
+		Status  string `json:"error"`
+		Message string `json:"message"`
+	}
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if arc == "" {
```

Comment:
more descriptive name of this variable would be appreciated, because first thing that came to my mind is that arc is either some service component or because it is string, then ARC URL.